### PR TITLE
Add support for SHA256 hashes in explicit files

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

- Comes from https://github.com/conda/ceps/pull/79#discussion_r1638083863
- Closes https://github.com/conda/conda/issues/7882
- Closes https://github.com/conda/conda/issues/2903 (added in 2016!)

I tried to make it backwards compatible, but older conda versions will fail with a SHA256. Hopefully the error message is easy to find so I'll mention it in the news file so Google takes folks the right place. Also here for good measure:

If you find an error like this:

```
conda.exceptions.ParseError: Could not parse explicit URL: https://conda.anaconda.org/conda-forge/linux-aarch64/ca-certificates-2024.7.4-hcefe29a_0.conda#562bfc2608d82996a08e5b5b2366ed319a51ace6a2518a004ba672edca75fc23
```

You presumably need to update `conda` to the latest version (>=24.9.0), which does have support for the 64-char long SHA256 hashes. I also added a `created-by:` comment to point out which tool and version generated the tool. Thought of adding something like `txtfile-version` or so on, but that'll require a bigger refactor if we want to utilize it.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/misc.py [hunk 1]
```diff
@@ -50,7 +50,10 @@ def conda_installed_files(prefix, exclude_self_build=False):
 url_pat = re.compile(
     r"(?:(?P<url_p>.+)(?:[/\\]))?"
     r"(?P<fn>[^/\\#]+(?:\.tar\.bz2|\.conda))"
-    r"(:?#(?P<md5>[0-9a-f]{32}))?$"
+    r"(:?#("
```
Comment (jaimergp) [inline conda/misc.py:53]:
```suggestion
    r"(?:#("
```

This must be a typo. I haven't ever seen a `<url>:#<md5>` lockfile.

File: conda/cli/main_list.py [hunk 2]
```diff
@@ -263,8 +273,14 @@ def print_explicit(prefix, add_md5=False, remove_auth=True):
...
         if remove_auth:
             url = common_url.remove_auth(common_url.split_anaconda_token(url)[0])
-        md5 = prefix_record.get("md5")
-        print(url + (f"#{md5}" if add_md5 and md5 else ""))
+        if add_md5:
+            md5 = prefix_record.get("md5")
+            print(url + (f"#{md5}" if md5 else ""))
+        elif add_sha256:
+            sha256 = prefix_record.get("sha256")
+            print(url + (f"#{sha256}" if sha256 else ""))
```
Comment (dholth) [inline conda/cli/main_list.py:281]:
```suggestion
        if add_md5 or add_sha256:
            hash_key = "md5" if add_md5 else "sha256"
            hash_value = prefix_record.get(hash_key)
            print(url + (f"#{hash_value}" if hash_value else ""))
```
Would this be code golf?

**Review by dholth — APPROVED** [review]
I like it

**Review by jezdez — DISMISSED** [review]

File: conda/cli/main_list.py [hunk 3]
```diff
@@ -263,8 +273,17 @@ def print_explicit(prefix, add_md5=False, remove_auth=True):
             continue
         if remove_auth:
             url = common_url.remove_auth(common_url.split_anaconda_token(url)[0])
-        md5 = prefix_record.get("md5")
-        print(url + (f"#{md5}" if add_md5 and md5 else ""))
+        if add_md5:
+            md5 = prefix_record.get("md5")
+            print(url + (f"#{md5}" if md5 else ""))
+        elif add_sha256:
+            sha256 = prefix_record.get("sha256")
```
Comment (jaimergp) [inline conda/cli/main_list.py:280]:
```suggestion
```

**Review by jezdez — APPROVED** [review]

**Pull request merged**